### PR TITLE
CPDNPQ-1257 Create NPQ Maths user flow on the service

### PR DESCRIPTION
**Pull request by javier-npq**
### Context

Ticket: https://dfedigital.atlassian.net/browse/CPDNPQ-1257

### Changes proposed in this pull request

- Add in option for Maths NPQ course selection page
- 3 new pages added just for the Maths NPQ 
- 2 question pages 
- 1 cannot take this course page 
- 1 different funding outcome message

### CODE REVIEW

**Comment by dalcJ** [comment]
1. Remove 'the' here: 
Screenshot 2023-08-11 at 11 27 52

2. link this text to https://www.gov.uk/guidance/join-the-maths-teaching-for-mastery-programme 
Screenshot 2023-08-11 at 11 27 56

@javier-npq

**Comment by dalcJ** [comment]
add 15px margin here: 

Screenshot 2023-08-11 at 11 32 32

@javier-npq

**Comment by dalcJ** [comment]
Add 2 links. 
'primary maths Teaching for Mastery programme' linked to https://www.gov.uk/guidance/join-the-maths-teaching-for-mastery-programme 

'register for a different NPQ' links back to the /registration/choose-your-npq page 

Screenshot 2023-08-11 at 11 34 31

@javier-npq

**Comment by dalcJ** [comment]
The back link on the 'Funding eligibility' page, after selecting either of these options is going to the choose an NPQ page, it should return to this page. 
 
Screenshot 2023-08-11 at 11 36 58

@javier-npq

**Comment by dalcJ** [comment]
The back link on the 'Funding eligibility' page, after selecting 'yes' is going to the choose an NPQ page, it should return to this page.

Screenshot 2023-08-11 at 11 38 46

 @javier-npq

**Comment by dalcJ** [comment]
The funding eligibility outcomes are not correct for when not eligible/ in review. 

These should work in the same way as the other NPQs. 

- Selecting 'No' to 'Do you work in England' = Not eligible. Should see this message: https://npq-prototype.herokuapp.com/funding/funding-not-available-england 
- Selecting Early years > private nursery as setting = not eligible. Should see this message: https://npq-prototype.herokuapp.com/funding/funding-not-available-setting 
- Selecting 'other' for setting = eligibility to be reviewed. Should see this message: https://npq-prototype.herokuapp.com/funding/edge-case

etc. 

@javier-npq

**Review by dalcJ — CHANGES_REQUESTED** [review]
See comments

**Comment by javier-npq** [comment]
> add 15px margin here:
> 
> Screenshot 2023-08-11 at 11 32 32
> 
> @javier-npq

@dalcJ That hint and options are displayed through a generic view for all pages so it will disturb other pages if we change the margin for this one. I'm afraid we can't touch this for now. But the rest of the changes have been implemented, thanks for a detailed review!

**Comment by dalcJ** [comment]
Thanks for making the changes Javier. Just a couple of bits. 

The colour of the links on here are not correct: 

Screenshot 2023-08-14 at 16 34 56

@javier-npq

**Comment by dalcJ** [comment]
> The funding eligibility outcomes are not correct for when not eligible/ in review.
> 
> These should work in the same way as the other NPQs.
> 
> * Selecting 'No' to 'Do you work in England' = Not eligible. Should see this message: https://npq-prototype.herokuapp.com/funding/funding-not-available-england
> * Selecting Early years > private nursery as setting = not eligible. Should see this message: https://npq-prototype.herokuapp.com/funding/funding-not-available-setting
> * Selecting 'other' for setting = eligibility to be reviewed. Should see this message: https://npq-prototype.herokuapp.com/funding/edge-case
> 
> etc.
> 
> @javier-npq

For this one, it's not working quite right. At the moment if the user is not eligible or eligibility is in review. And then they select Primary mathematics they are shown the eligibility outcome as the next page. 

But it should instead take the user to the additional questions that we've added for primary mathematics. They would just see the correct eligibility different message after completing those questions. Let me know if you want to chat over this on a call, might be easier to go through the flow together. 
@javier-npq

**Comment by dalcJ** [comment]
> > add 15px margin here:
> > Screenshot 2023-08-11 at 11 32 32
> > @javier-npq
> 
> @dalcJ That hint and options are displayed through a generic view for all pages so it will disturb other pages if we change the margin for this one. I'm afraid we can't touch this for now. But the rest of the changes have been implemented, thanks for a detailed review!

@javier-npq can't the class `govuk-!-margin-top-3` be added to the radios just on this page?

File: app/views/registration_wizard/possible_funding/_funding_eligible_maths.html.erb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<h1 class="govuk-heading-xl">Funding eligibility</h1>
+
+<p class="govuk-body">
+  If your provider accepts your application, you'll be eligible for scholarship funding for the <%= localise_sentence_embedded_course_name(course) %>.
```
Comment (EmmaFrith) [inline app/views/registration_wizard/possible_funding/_funding_eligible_maths.html.erb:4]:
```suggestion
  If your provider accepts your application, you’ll be eligible for scholarship funding for the <%= localise_sentence_embedded_course_name(course) %>.
```

File: app/views/registration_wizard/maths_cannot_register.html.erb [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+    'registration_wizard/shared/copy_page',
+    form: @form,
+    wizard: @wizard,
+    ) do
+%>
+  <h1 class="govuk-heading-xl">You cannot register for the leading primary mathematics NPQ</h1>
+
+  <p class="govuk-body">This is because you said that you cannot show an understanding of mastery approaches to teaching maths.</p>
+
+  <p class="govuk-body">Join the <a href="https://www.gov.uk/guidance/join-the-maths-teaching-for-mastery-programme">primary maths Teaching for Mastery programme</a> or go back if you want to <a href="<%= registration_wizard_show_url('choose_your_npq'.dasherize) %>">register for a different NPQ.</a></p>
```
Comment (EmmaFrith) [inline app/views/registration_wizard/maths_cannot_register.html.erb:12]:
```suggestion
  <p class="govuk-body">Join the <a href="https://www.gov.uk/guidance/join-the-maths-teaching-for-mastery-programme">primary maths Teaching for Mastery programme</a> or go back if you want to <a href="<%= registration_wizard_show_url('choose_your_npq'.dasherize) %>">register for a different NPQ</a>.</p>
```

**Comment by dalcJ** [comment]
@javier-npq  

Also, the route for non-state funded schools is showing as eligible. This should show as not eligible due to setting: 

Screenshot 2023-08-15 at 15 07 22

File: app/lib/forms/maths_eligibility_teaching_for_mastery.rb [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+      [
+        build_option_struct(value: "yes", link_errors: true),
+        build_option_struct(value: "no", hint: I18n.t("helpers.hint.registration_wizard.maths_eligibility_teaching_for_mastery_hint")),
+      ]
+    end
+
+    def next_step
+      if maths_eligibility_teaching_for_mastery == "yes"
+        wizard.store["maths_understanding"] = true
+        if !wizard.query_store.teacher_catchment_england? || wizard.query_store.kind_of_nursery_private?
```
Comment (slawosz) [inline app/lib/forms/maths_eligibility_teaching_for_mastery.rb:35]: You have here 2 branches with the same outcome but actually I really like it as the branching questions are grouped onto logical chunks.

File: spec/features/happy_journeys/js/while_working_at_private_childcare_provider_but_not_a_nursery_spec.rb [hunk 4]
```diff
@@ -66,21 +66,28 @@
...
 
     ineligible_courses.each do |course|
       expect_page_to_have(path: "/registration/choose-your-npq", submit_form: true) do
         expect(page).to have_text("Which NPQ do you want to do?")
         page.choose(course, visible: :all)
       end
 
-      expect(page).to have_text("you are eligible for scholarship funding for")
-      expect(page).to have_text("You can go back and select the Early years leadership")
+      if Flipper.enabled?(:maths_npq)
```
Comment (slawosz) [inline spec/features/happy_journeys/js/while_working_at_private_childcare_provider_but_not_a_nursery_spec.rb:83]: Can't Flipper be enabled by default in tests?

**Review by slawosz — APPROVED** [review]

**Review by dalcJ — APPROVED** [review]

**Comment by peteryates** [comment]
Regarding the margins, shouldn't we just stick to the ones provided by the design system? Is there a slight issue with our markup that's affecting them - they do look smaller than usual (and the legend is very big)

**Comment by javier-npq** [comment]
> Regarding the margins, shouldn't we just stick to the ones provided by the design system? Is there a slight issue with our markup that's affecting them - they do look smaller than usual (and the legend is very big)

We are using the styling/css provided by the design system. They are present on selected pages as far as i have seen, will be raising a separate ticket where I can have a look at the root cause of this and fix it.

**Pull request merged**